services/blockchain-indexer/events/blockchain.js

```diff
@@ -80,6 +80,38 @@ module.exports = [
 			Signals.get('newBlock').add(newTransactionsListener);
 		},
 	},
+	{
+		name: 'block.delete',
+		description: 'Emit the deleted block.',
+		controller: callback => {
+			const deleteBlockListener = async (data) => {
+				try {
+					if (data && Array.isArray(data.data)) {
+						callback(data);
+					}
+				} catch (err) {
+					logger.error(`Error occured when processing 'block.delete' event:\n${err.stack}`);
+				}
+			};
+			Signals.get('deleteBlock').add(deleteBlockListener);
```

Comment:
Can we rename the event payload to `payload` instead of `data`? Please consider doing the same for the remaining events. Also, please consider adding debug logs for all of them as well.
```suggestion
			const deleteBlockListener = async (payload) => {
				try {
					if (payload && Array.isArray(payload.data)) {
						callback(payload);
					} else {
						const payloadStr = JSON.stringify(payload);
						logger.debug(`Incorrect payload detected for 'deleteBlock' signal:\n${payload}`);
					}
				} catch (err) {
					logger.error(`Error occured when processing 'block.delete' event:\n${err.stack}`);
				}
			};
			Signals.get('deleteBlock').add(deleteBlockListener);
```